### PR TITLE
Stats: add start date for devices and utm stats

### PR DESCRIPTION
**Pull request by kangzj**
Related to https://github.com/Automattic/red-team/issues/230

https://github.com/Automattic/red-team/issues/239

https://github.com/Automattic/red-team/issues/240

## Proposed Changes

* Add start_date when querying devices and utm endpoints

## Why are these changes being made?

* Date range support

## Testing Instructions

* Open Calypso Live Branch and apply feature flag `flags=stats/new-date-filtering`
* Ensure the default date range is reflect for modules `UTM` and `Devices` work well
* Choose another date range ensure the three modules still work well
* Inspect network requests
* Ensure correct params are passed to the back end
* Click 'view all'
* Ensure summary page works well
* Ensure the traffic page still works as is without the feature flag

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/hooks/utils.ts [hunk 1]
```diff
@@ -27,8 +29,10 @@ export const processQueryParams = ( query: QueryStatsParams ) => {
 	// `max` is probably set to 0 to fetch all results.
 	const max = query.max ?? 10;
 	const date = query.date || new Date().toISOString().split( 'T' )[ 0 ];
+	// Explicitly test start_date?
 
 	// Calculate the number of days to query based on the period.
+	// What is days for?? 🧐
```
Comment (kangzj) [inline client/my-sites/stats/hooks/utils.ts]:
```suggestion
```

File: client/my-sites/stats/hooks/utils.ts [hunk 2]
```diff
@@ -27,8 +29,10 @@ export const processQueryParams = ( query: QueryStatsParams ) => {
 	// `max` is probably set to 0 to fetch all results.
 	const max = query.max ?? 10;
 	const date = query.date || new Date().toISOString().split( 'T' )[ 0 ];
+	// Explicitly test start_date?
```
Comment (kangzj) [inline client/my-sites/stats/hooks/utils.ts]:
```suggestion
```

File: client/my-sites/stats/hooks/utils.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 export interface QueryStatsParams {
 	date?: string;
+	start_date?: string;
 	days?: number;
 	max?: number;
 	num?: number;
-	period?: string;
+	period?: 'day' | 'week' | 'month' | 'year';
```
Comment (kangzj) [inline client/my-sites/stats/hooks/utils.ts]:
```suggestion
	period?: string;
```

**Comment by a8ck3n** [comment]
I'm seeing the `start_date` param passed even when the flag is disabled. It doesn't appear to break anything but just wanted to flag it in case this wasn't intended.

```
https://public-api.wordpress.com/rest/v1.1/sites/20115252/stats/devices/screensize?http_envelope=1&period=day&date=2024-10-28&start_date=&num=1&max=10&days=1
```

**Pull request merged**